include/sys/spa_impl.h

```diff
@@ -425,6 +425,7 @@ struct spa {
 	kcondvar_t	spa_waiters_cv;
 	int		spa_waiters;		/* number of waiting threads */
 	boolean_t	spa_waiters_cancel;	/* waiters should return */
+	boolean_t	spa_raidz_expanding;	/* expansion in progress */
```

Comment:
Why can't we use `spa_raidz_expand != NULL`, rather than adding a new field?  Are they set/cleared at different times?  It looks to me like `spa_raidz_expand` and `spa_raidz_expanding` are both set while the namespace lock is held, so don't think there's any race condition when the expansion starts.

Reply:
Because since the spa_raidz_expand pointer is set in Syncing context which is run in a different task AIUI.  we would have a race where we could exit here and another expand request could get in here and also start an expand before the pointer gets set.  I wanted to use the pointer setting but this seemed better.  (Unless I am misunderstanding how things work)

Reply:
Yeah so it happens via:
```
		dmu_tx_t *tx = dmu_tx_create_assigned(spa->spa_dsl_pool, txg);
		dsl_sync_task_nowait(spa->spa_dsl_pool, vdev_raidz_attach_sync,
		    newvd, 0, ZFS_SPACE_CHECK_EXTRA_RESERVED, tx);
```
which means that it will happen when that txg is synced.  Which we wait for with the call to `spa_vdev_exit(spa, newrootvd, dtl_max_txg, 0);`.  It looks like this all happens while holding the `spa_namespace_lock`, so another removal couldn't be initiated before we set `spa_raidz_expand`.

Reply:
Ah, I missed that spa_vdev_exit() waited till the TXG is finished.  That makes it simpler, will change to use the expand pointer.